spotifycli/test/dbus_metadata_response.py

```diff
@@ -0,0 +1,14 @@
+if args[1] == 'Metadata':
+    ret = {
+        'xesam:artist': ['', 'The Beatles'],
+        'xesam:title': ['', 'Yesterday'],
+        'xesam:album': ['', 'Help!'],
+        'mpris:length': ['', '10000000'],
+        'mpris:artUrl': ['', 'https://github.com/pwittchen/spotify-cli-linux'],
+    }
+elif args[1] == 'PlaybackStatus':
+    ret = 'Playing'
+elif args[1] == 'Position':
+    ret = 1000000
+else:
+    ret = args
```

Comment:
OK.  Here's the meat of the testing work: building out a pretend dbus response based on the lookup argument. 

## What is `ret`?

When you mock up dbus and provide a fake interface, you need to provide some Python code to be executed within the context of an existing Python template.  In this template, it will return the value of the variable `ret`, so the main outcome here is setting that variable.  

## What is `args`?

As with `ret`, in the dbus template, `args` is passed in as the body of the dbus method call.  Just grab the index of interest and see what the dbus call was, and decide what to return.

## Marshaling data

Dbus seems to be exceedingly annoying about the particular types being returned.  For example, when you look up `Metadata`, it returns a dictionary of key value pairs.  The values are all lists, where the first item is empty and the second item is the meaningful value.  Note that the meaningful value here is always a string, even when it's a numeric like the `mpris:length`.  And when you run dbus locally, the `mpris:artist` actually returns a list of artists in that meaningful item.  

**But**, if I try to provide a slight variation of types on any of the meaningful items there, the dbus marshalling functions fail aggressively.  I would love to mimic the way it actually works on my computer by instead of saying `'The Beatles'`, I could say `['The Beatles', ]`.  But unfortunately I can't mix that with the other types.  And if I try to use a numeric value for `mpris:length` instead of a string number, it also fails.  So annoying.

The root cause is that the dbus Python package is trying to detect the types dynamically so that it can marshal the data into the right underlying types to perform dbus communication.  Dbus does allow dicts and objects, floats, ints, and bools, and more.  But when you pass the data in, you need to tell Dbus exactly what each data member is. 

Anyway, this little bit of annoyance is what caused the minor tweaks to the spotifycli.py file above to handle data coming in as a list/string or as a int/string.

## Best path forward

I learned more about dbus today than I ever intended.  The best way forward would be if someone else who knew a little bit more could come along and help me specify these return values better, we could revert my little tweaks to the library above.  And that would be great.  I'll keep trying as well...